### PR TITLE
Add generate credentials

### LINKED ISSUE #2: Add generate credentials command
The generate credentials command generates a new Ed25519 key pair and associated DID. Let's add a implementation for this command. 🦀

### PR DESCRIPTION
**Pull request by BoisterousCoder**
This PR implements the add generate credentials command.

#### Testing
To test this command 

```sh 
cargo run generate credentials
```

The output should produce the ed222519 key pair and its did, which should look something like the following:

```
✅ Generated an Ed25519 key pair and associated DID
🗝️ Private key: uwBK0fIWYZeg8Dr5gyw8/8Yf0khRxFegIPeWgP2o5jI=
🔑 Public key: ANqzyN5Y33byvfbf6Z7J54Jo1ZQn978YFrIIZbMrSnI=
🆔 DID: did:key:z6M14LRScqxd9a9XWJko3toZy9xQbHxWEo8SCCfy5PRCGFs
```

#### Related Issues

Closes #2

### CODE REVIEW

File: src/cmd/generate.rs [hunk 1]
```diff
@@ -14,7 +20,56 @@ pub enum GenerateCommands {
...
         GenerateCommands::Credentials => {
-            todo!("credentials")
+            let private_key = SigningKey::new(thread_rng());
+
+            let (private_key_str, public_key_str, did) = generate_key_strs(private_key);
+
+            println!("{}", "✅ Generated an Ed25519 key pair and associated DID".bright_green());
+            println!("🗝️  Private key: {}", private_key_str.bright_blue());
+            println!("🔑 Public key: {}", public_key_str.bright_blue());
+            println!("🆔 DID: {}", did.bright_blue());
```
Comment (bgins) [inline src/cmd/generate.rs]: Nice job capturing the emoji! 😄

**Review by bgins — APPROVED** [review]
Looks great! Thanks for implementing this. 🎉 🚀

File: src/cmd/generate.rs [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+use colored::Colorize;
 
 #[derive(Args)]
 pub struct Generate {
     #[clap(subcommand)]
     command: GenerateCommands,
 }
 
+const DID_KEY_PREFIX:&str = "did:key:z";
+const DID_KEY_PREFIX_ED25519:[u8; 2] = [237, 1];
```
Comment (expede) [inline src/cmd/generate.rs]: Oh interesting! The existing did-key crates don't have Ed25519?

Comment (BoisterousCoder) [reply src/cmd/generate.rs]: The one did crate I saw had no documentation and looked unfinished, so I just made my own as it was pretty quick

**Pull request merged**